packages/vite/src/node/plugins/html.ts

```diff
@@ -196,6 +213,10 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
                 js += `\nimport "${id}?html-proxy&index=${inlineModuleIndex}.js"`
                 shouldRemove = true
               }
+
+              everyScriptIsAsync &&= isAsync
+              someScriptsAreAsync ||= isAsync
+              someScriptsAreDefer ||= !isAsync
```

Comment:
![giphy](https://c.tenor.com/QadXaFWD-j0AAAAC/cat-wow.gif)

Reply:
Is this a good or a bad reaction? 😅
We may need to refactor this info as an array and then at the end just use every/some over it, if thus is not clear

Reply:
It is a good reaction 🤣 It was just like a 😲
I also approved the PR 😸